### ISSUE TITLE
Change default font of build event text boxes to monospace

### ISSUE THREAD

**Issue by asbjornu** [dotnet/roslyn-project-system#189]
As mentioned in #176, it would be nice if the default font for the text boxes in the "Build events" page were monospace.

**Comment by davkean** [comment]
We probably shouldn't explicitly set the font, and instead probably use the Fonts settings from [All Text Tool Windows].

**Comment by asbjornu** [comment]
@davkean Would that setting affect both single- and multi-line text boxes? Because it's only the multiline text boxes I think makes sense to have monospaced.

**Comment by davkean** [comment]
Yeah, just multi-line text boxes.

**Comment by asbjornu** [comment]
Ok, sounds good. Can that font setting please be monospace by default, then? :wink:

**Comment by davkean** [comment]
It is:
![image](https://cloud.githubusercontent.com/assets/1103906/15550813/08e3810c-2268-11e6-9154-312f3ada9be3.png)

**Comment by davkean** [comment]
Unsure if that's the right setting to use, but don't think it's worth creating a new one just for these text boxes.

**Comment by asbjornu** [comment]
I agree a new settings isn't required. And since `Consolas` is the default font already, it's just a matter of applying this font to the multiline text boxes. Awesome! 😄

**Comment by eriawan** [comment]
@davkean 
This issue should be closed, because there's no work to do anymore. I can safely assume it's resolved 🙂

**Comment by davkean** [comment]
@eriawan not sure I follow? This isn't resolved, the build event text box is not monospace yet.

**Comment by eriawan** [comment]
@davkean I mean, based on your suggestion that new setting isn't required and I thought it was solved. 
My bad, I misunderstood the context, apologize for this.

FYI, I have tried the [All Text Tool Windows] and the font isn't changed.

**Comment by davkean** [comment]
@eriawan, yes that's exactly what this bug is representing; make a change to the Build events text box to respect that setting. :)

**Comment by eriawan** [comment]
@davkean I really need to know how the C#/VB/F# editor get the font settings from **Font & Colors** Option dialog. Just a suggestion, maybe I can hook the font setting of "Build event" tab's multiline text to use Text Editor's font and size from Font & Colors. 

Where is the code that does this checking of Font setting for C#/VB/F# text (code) editor?

**Comment by eriawan** [comment]
@davkean 
Just a humble reminder 🙂 
Could you help me to locate the code that check the font setting of C#/VB/F# code editor? I'm going to have this as starting point to enable setting the font of Build Event text editor.

**Comment by davkean** [comment]
To be honest, I don't know - I've never done it myself. I'd start searching in [Roslyn](http://github.com/dotnet/roslyn) - I'm sure they've done it.

**Comment by eriawan** [comment]
@davkean 
Looks like I can use the **Text Editor**  category from Fonts and Colors setting, by getting the setting from IVsFontAndColorStorage. I can also hook to the font change event by using `TextManagerEvents` class.

Could you tell me how to debug the project that contains the Build Events, this https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj ?

**Comment by eriawan** [comment]
@davkean 
Just a humble reminder again 🙂 
Looks like I can use the Text Editor category from Fonts and Colors setting, by getting the setting from IVsFontAndColorStorage. I can also hook to the font change event by using TextManagerEvents class.

Could you tell me how to debug the project that contains the Build Events, this https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj ?

**Comment by davkean** [comment]
Sorry, for the late reply. To debug that code, simply follow these instructions: https://github.com/dotnet/project-system/blob/master/docs/repo/getting-started.md#visual-studio-1 and then create any C#/VB project.

**Comment by davkean** [comment]
And IVsFontAndColorStorage does look promising, OpenCategory, then GetFont.

**Comment by eriawan** [comment]
@davkean 
My PR #2499 has been submitted. Please review 🙂
